wct-core/src/main/java/org/webcurator/ui/target/validator/ProfilesOverridesValidator.java

```diff
@@ -65,8 +65,11 @@ public void validate(Object comm, Errors errors) {
 			} else {
 				log.info("Skipping imported profile validation as there is no profile override.");
 			}
-		} else {
-			log.info("Validating non-imported profile.");
+		}
+		// Only validate Heritrix 1 fields if a Heritrix 1 profile is selected.
+		// (prevents hidden H1 fields failing validation when using an H3 profile)
+		else if (command.getHarvesterType().equals("HERITRIX1")){
```

Comment:
Just a little trick when comparing strings: always use the constant first, so:
`else if ("HERITRIX1".equals(command.getHarvesterType()) {`
This means that you don't have to worry about nulls.

Reply:
Will change